packages/design/src/DataTable/StyledTable.tsx

```diff
@@ -115,3 +115,41 @@ export const StyledEmptyIndicator = styled.div(
   }
 `
 );
+
+const labelWidths = css`
+  @media (max-width: 1300px) {
+    max-width: 40em;
+  }
+
+  @media (max-width: 1100px) {
+    max-width: 30em;
+  }
+`;
+
+export const LabelContent = styled.div`
+  line-height: 20px;
+  position: relative;
+  top: 1px;
+
+  ${labelWidths}
+
+  ${({ dots }) => {
+    if (dots) {
+      return css`
+        &:after {
+          content: '...';
+        }
+      `;
+    }
+  }}
+`;
+
+export const LabelWrapper = styled.div`
+  display: flex;
+  flex-direction: row;
+  flex-wrap: wrap;
+  max-width: 100%;
+  align-items: center;
+
+  ${labelWidths}
+`;
```

Comment:
unless i'm missing something, this is not necessary. CSS has an [ellipsis](https://www.w3schools.com/cssref/tryit.php?filename=trycss3_text-overflow) feature

Reply:
@kimlisa Thanks for noticing it. I agree and I tried to use it at first, but I found difficulties that I have described below. 

This feature does not work correctly here. We would have to set the dimensions on the label for it to work here. 
And given the specificity of the usage of labels:
1. It is used in different tables that have a different amount of columns and content in them
2. Number of labels and text length of them can vary even in the same table. 
3. They can be used on many different screens and devices.

I found it difficult to use that feature to correctly work in a lot of different use cases. Also given that we want to display more or less 100 chars of the label, I found it was just easier to shorten the label with JS and add dots with CSS after the element (so it cannot be selected). 

It is not a perfect solution, but it works a lot better because:
1. Labels can adjust their dimensions to match the available space and length of their content. 
2. It is simple in implementation compared to juggling a lot of CSS to make the ``ellipsis`` feature work as we want it to work.

I just took an approach that looked simpler. I always try to stick to the KISS principle when possible. I have tried to use the ellipsis CSS feature before and found the difficulties described above. 

I hope that answers your question.

Reply:
To better illustrate the issue I made a [pen](https://codepen.io/sielakos/pen/dyKjqMe) 

Those are basically some divs with an ellipsis inside a table. Which is our use case.

1. So the red case of not setting a dimension for the div does not allow for the ellipsis feature to work. 

2. The green case with ``max-width`` set to some value enables ellipsis, but on a smaller screen it still does not work the table blows horizontally and we observe the horizontal scroll bar of death. 
Also, it would be difficult to set ``max-width`` that works well with multiple different tables.

3. The blue case with text wrapping enabled does not cause the horizontal bar to appear, but also ellipsis does not work and we see full text. We would have to set ``max-height`` here for it to work. And getting it to display 100 chars of the label and work correctly in multiple conditions would be a challenge. 

<img width="793" alt="image" src="https://user-images.githubusercontent.com/141401/204504088-b3a3e0d3-e77b-4d4a-8978-c9d465f5b024.png">

That's why just cutting text is an easier (and therefore better) solution in my opinion. 

All that said, I welcome all better solutions. It's just that I have experimented a lot to make it match 2 criteria:
1. Be responsive
2. Show at max 100 chars of the label, but not less than that if the label is longer. 

And came to the conclusion that a bit of fairly simple JS (it is just the ``.substring`` call after all) is better than a large block of magical CSS.

Reply:
can you try this out https://github.com/gravitational/webapps/commit/454c6aec11928e00dc80c06c67bdea6419b3980b, and let me know what you think.

keep in mind our app is not designed for really small screens like mobile phone